plugins-compat-tester/src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -539,39 +539,67 @@ private TestExecutionResult testPluginAgainst(MavenCoordinates coreCoordinates,
     }
 
     private void addSplitPluginDependencies(String thisPlugin, MavenRunner.Config mconfig, File pluginCheckoutDir, MavenPom pom, Map<String,Plugin> otherPlugins, Map<String, String> pluginGroupIds) throws PomExecutionException, IOException {
+        System.out.println("STARTING PLUGIN DEPS");
         File tmp = File.createTempFile("dependencies", ".log");
         VersionNumber coreDep = null;
         Map<String,VersionNumber> pluginDeps = new HashMap<String,VersionNumber>();
+        Map<String,VersionNumber> pluginDepsTest = new HashMap<String,VersionNumber>();
         try {
             runner.run(mconfig, pluginCheckoutDir, tmp, "dependency:resolve");
             Reader r = new FileReader(tmp);
             try {
                 BufferedReader br = new BufferedReader(r);
-                // TODO could include |test but only if pom.addDependencies would add as <scope>test</scope>
                 Pattern p = Pattern.compile("\\[INFO\\]    ([^:]+):([^:]+):([a-z-]+):([^:]+):(provided|compile|runtime|system)");
+                Pattern p2 = Pattern.compile("\\[INFO\\]    ([^:]+):([^:]+):([a-z-]+):([^:]+):(test)");
                 String line;
                 while ((line = br.readLine()) != null) {
                     Matcher m = p.matcher(line);
-                    if (!m.matches()) {
-                        continue;
-                    }
-                    String groupId = m.group(1);
-                    String artifactId = m.group(2);
+                    Matcher m2 = p2.matcher(line);
+                    String groupId;
+                    String artifactId;
                     VersionNumber version;
-                    try {
-                        version = new VersionNumber(m.group(4));
-                    } catch (IllegalArgumentException x) {
-                        // OK, some other kind of dep, just ignore
+                    if (!m.matches() && !m2.matches()) {
                         continue;
+                    } else if (m.matches()) {
```

Comment:
Maybe this logic could be extracted into a method, as it's the same in both if clauses.